src/main/java/seedu/storemando/model/item/LocationContainsPredicate.java

```diff
@@ -0,0 +1,27 @@
+package seedu.storemando.model.item;
+
+import java.util.function.Predicate;
+
+/**
+ * Tests that a {@code Item}'s {@code location} matches any of the keywords given.
+ */
+public class LocationContainsPredicate implements Predicate<Item> {
+    private final String keyword;
+
+    public LocationContainsPredicate(String keyword) {
+        this.keyword = keyword;
+    }
+
+    @Override
+    public boolean test(Item item) {
+        return keyword.equals(item.getLocation().value);
```

Comment:
i think this won't work if item location is like "Kums     room" and the command is "clear Kums room". Maybe you can try it out and see if it works.

Reply:
the item location has a lot of white spaces in between Kums and room.

Reply:
Yes, I think this would not work with additional spaces as well using equals.

Reply:
Yes, this will not work, and it is supposed to be that way. This follows the "add" command, where location "bed <lots of whitespaces> room" is considered a different location from "bed room". Hence, when I clear items in location "bed <lots of whitespaces> room", items in location "bed room" will not be deleted.

![image](https://user-images.githubusercontent.com/57126871/112756431-b62d4600-9017-11eb-8c13-f2971eb09290.png)